library/std/src/sys/windows/thread.rs

```diff
@@ -87,7 +88,17 @@ impl Thread {
     }
 
     pub fn sleep(dur: Duration) {
-        unsafe { c::Sleep(super::dur2timeout(dur)) }
+        fn high_precision_sleep(dur: Duration) -> Result<(), ()> {
+            let timer = WaitableTimer::high_resolution()?;
```

Comment:
How good is the performance of `CreateWaitableTimerExW`? If it is slow, it might be better to cache the timer in a thread local.

Reply:
It's around 2,000 nanoseconds on my machine (minus the time it takes to call `QueryPerformanceCounter`). That is well within noise for waiting on my machine. Granted, other hardware may be able to perform better but I'd rather avoid adding a thread local unless or until someone can show it's worth it.